core/perf_test/test_sharedSpace.cpp

```diff
@@ -0,0 +1,343 @@
+#elif defined KOKKOS_ENABLE_HIP
+  return Kokkos::Experimental::HIP::hip_device_prop().totalGlobalMem;
+#elif defined KOKKOS_ENABLE_SYCL
+  auto device = Kokkos::Experimental::SYCL{}.sycl_queue().get_device();
+  return device.get_info<sycl::info::device::global_mem_size>();
+#else
+#error \
+    "The sharedMemory test is only defined for Kokkos::Cuda, Kokkos::HIP, and Kokkos::SYCL"
+  return 0;
+#endif
+}
+
+struct Arguments {
+  unsigned int numRepetitions       = 10;
+  unsigned int numWarmupRepetitions = 100;
+  unsigned int numDeviceHostCycles  = 3;
+  double fractionOfDeviceMemory     = 0.4;
+  double threshold                  = 2.0;
+};
+
+void test_sharedSpace(Arguments args) {
+  const unsigned int numRepetitions       = args.numRepetitions;
+  const unsigned int numWarmupRepetitions = args.numWarmupRepetitions;
+  const unsigned int numDeviceHostCycles  = args.numDeviceHostCycles;
+  double fractionOfDeviceMemory           = args.fractionOfDeviceMemory;
+  double threshold                        = args.threshold;
+  size_t numBytes = fractionOfDeviceMemory * getDeviceMemorySize();
+  size_t numPages = numBytes / getBytesPerPage();
+
+  // ALLOCATION
+  Kokkos::View<int*, Kokkos::SharedSpace> migratableData(
+      "migratableData", numPages * getBytesPerPage() / sizeof(int));
+  Kokkos::View<int*, Kokkos::DefaultExecutionSpace::memory_space> deviceData(
+      "deviceData", numPages * getBytesPerPage() / sizeof(int));
+  Kokkos::View<int*, Kokkos::DefaultHostExecutionSpace::memory_space> hostData(
+      "hostData", numPages * getBytesPerPage() / sizeof(int));
+  Kokkos::fence();
+
+  // WARMUP GPU
+  incrementInLoop<Kokkos::DefaultExecutionSpace>(
+      deviceData,
+      numWarmupRepetitions);  // warming up gpu
+
+  // GET DEVICE LOCAL TIMINGS
+  auto deviceLocalResults = incrementInLoop<Kokkos::DefaultExecutionSpace>(
+      deviceData, numRepetitions);
+
+  // WARMUP HOST
+  incrementInLoop<Kokkos::DefaultHostExecutionSpace>(
+      hostData,
+      numWarmupRepetitions);  // warming up host
+  // GET HOST LOCAL TIMINGS
+  auto hostLocalResults = incrementInLoop<Kokkos::DefaultHostExecutionSpace>(
+      hostData, numRepetitions);
+
+  // GET PAGE MIGRATING TIMINGS DATA
+  std::vector<decltype(deviceLocalResults)> deviceResults{};
+  std::vector<decltype(hostLocalResults)> hostResults{};
+  for (unsigned i = 0; i < numDeviceHostCycles; ++i) {
+    // WARMUP GPU
+    incrementInLoop<Kokkos::DefaultExecutionSpace>(
+        deviceData,
+        numWarmupRepetitions);  // warming up gpu
+    // GET RESULTS DEVICE
+    deviceResults.push_back(incrementInLoop<Kokkos::DefaultExecutionSpace>(
+        migratableData, numRepetitions));
+
+    // WARMUP HOST
+    incrementInLoop<Kokkos::DefaultHostExecutionSpace>(
+        hostData,
+        numWarmupRepetitions);  // warming up host
+    // GET RESULTS HOST
+    hostResults.push_back(incrementInLoop<Kokkos::DefaultHostExecutionSpace>(
+        migratableData, numRepetitions));
+  }
```

Comment:
Maybe I'm reading things wrong, but shouldn't the warmup calls here both access `migratableData` - i.e. pull the pages to the space being measured?

Or is the warmup being performed here the clock-speed warmup, to make sure that each core is running at full speed? If so, please elaborate the comments within this loop to clarify.

Reply:
It is indeed to ensure the core-clock is at max when we do the actual measurement that does include the page migration. Therefore, it should not use the `migratableData`

Reply:
hope [3feb4d](https://github.com/kokkos/kokkos/pull/5289/commits/3feb4de9b4700b0d31b4df2e0347c77487aeda20) is helping